implementation/cdi/src/main/java/io/smallrye/opentelemetry/implementation/cdi/OpenTelemetryProducer.java

```diff
@@ -31,30 +33,72 @@
 import io.smallrye.opentelemetry.api.OpenTelemetryBuilderGetter;
 import io.smallrye.opentelemetry.api.OpenTelemetryConfig;
 import io.smallrye.opentelemetry.api.OpenTelemetryHandler;
 
 @Singleton
 public class OpenTelemetryProducer {
+
+    // n.b. while the security manager is supported by this project, take extra caution when evolving this class
+    // since otel initialization does require a lot of different file/logging/runtime/management security permissions
+
     private final List<AutoCloseable> closeables = new ArrayList<>();
 
     @Produces
     @Singleton
     public OpenTelemetry getOpenTelemetry(final OpenTelemetryConfig config) {
-        OpenTelemetry openTelemetry = new OpenTelemetryBuilderGetter().apply(config)
-                .disableShutdownHook()
-                .build()
-                .getOpenTelemetrySdk();
+        AutoConfiguredOpenTelemetrySdkBuilder autoConfiguredOpenTelemetrySdkBuilder = new OpenTelemetryBuilderGetter()
+                .apply(config).disableShutdownHook();
+
+        AutoConfiguredOpenTelemetrySdk otelSdk;
+        if (System.getSecurityManager() == null) {
+            otelSdk = autoConfiguredOpenTelemetrySdkBuilder.build();
+        } else {
+            // Requires FilePermission/RuntimePermission
+            otelSdk = AccessController.doPrivileged(
+                    (PrivilegedAction<AutoConfiguredOpenTelemetrySdk>) autoConfiguredOpenTelemetrySdkBuilder::build);
+        }
+
+        OpenTelemetry openTelemetry = otelSdk.getOpenTelemetrySdk();
 
         closeables.addAll(Classes.registerObservers(openTelemetry));
-        closeables.addAll(Cpu.registerObservers(openTelemetry));
+
+        // Requires FilePermission/RuntimePermission/ManagementPermission
+        List<AutoCloseable> cpuObservers;
+        if (System.getSecurityManager() == null) {
+            cpuObservers = Cpu.registerObservers(openTelemetry);
+        } else {
+            cpuObservers = AccessController
+                    .doPrivileged((PrivilegedAction<List<AutoCloseable>>) () -> Cpu.registerObservers(openTelemetry));
+        }
+        closeables.addAll(cpuObservers);
+
         closeables.addAll(MemoryPools.registerObservers(openTelemetry));
         closeables.addAll(Threads.registerObservers(openTelemetry));
-        closeables.addAll(GarbageCollector.registerObservers(openTelemetry));
 
-        OpenTelemetryHandler.install(openTelemetry);
+        List<AutoCloseable> garbageCollectorObservers;
+        if (System.getSecurityManager() == null) {
+            garbageCollectorObservers = Cpu.registerObservers(openTelemetry);
+        } else {
+            // Requires FilePermission/RuntimePermission
+            garbageCollectorObservers = AccessController.doPrivileged(
+                    (PrivilegedAction<List<AutoCloseable>>) () -> GarbageCollector.registerObservers(openTelemetry));
+        }
+        closeables.addAll(garbageCollectorObservers);
+
+        if (System.getSecurityManager() == null) {
```

Comment:
Can all of these SM checks be reduced to one and have all the work done together? The repeated checks makes the code a bit noisy...

Reply:
Something like this should work, no?

```
        // Requires FilePermission/RuntimePermission/ManagementPermission
        List<AutoCloseable> cpuObservers;
        List<AutoCloseable> garbageCollectorObservers;

        if (System.getSecurityManager() == null) {
            cpuObservers = Cpu.registerObservers(openTelemetry);
            garbageCollectorObservers = Cpu.registerObservers(openTelemetry);
        } else {
            cpuObservers = AccessController
                    .doPrivileged((PrivilegedAction<List<AutoCloseable>>) () -> Cpu.registerObservers(openTelemetry));
            garbageCollectorObservers = AccessController.doPrivileged(
                    (PrivilegedAction<List<AutoCloseable>>) () -> GarbageCollector.registerObservers(openTelemetry));
        }

        closeables.addAll(cpuObservers);
        closeables.addAll(MemoryPools.registerObservers(openTelemetry));
        closeables.addAll(Threads.registerObservers(openTelemetry));
        closeables.addAll(garbageCollectorObservers);

        if (System.getSecurityManager() == null) {
            OpenTelemetryHandler.install(openTelemetry);
        } else {
            // Requires LoggingPermission
            AccessController.doPrivileged((PrivilegedAction<List<AutoCloseable>>) () -> {
                OpenTelemetryHandler.install(openTelemetry);
                return null;
            });
        }
```

Reply:
OK, if `#registerObservers(OpenTelemetry)` has no side-effects, we can don that. Done.